lib/web/integrations_awsoidc.go

```diff
@@ -857,7 +859,22 @@ func (h *Handler) awsOIDCConfigureIdP(w http.ResponseWriter, r *http.Request, p
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
-	proxyAddr, err := oidc.IssuerFromPublicAddress(h.cfg.PublicProxyAddr)
+	s3Bucket := queryParams.Get("s3Bucket")
+	s3Prefix := queryParams.Get("s3Prefix")
+	if s3Bucket == "" || s3Prefix == "" {
+		return nil, trace.BadParameter("s3Bucket and s3Prefix query params are required")
+	}
+	s3Location := fmt.Sprintf("s3://%s/%s", s3Bucket, s3Prefix)
+	s3URI, err := url.Parse(s3Location)
```

Comment:
Do we know what a valid bucket name and object name prefix look like? Maybe we can check that and then build a URL instead.

Reply:
Not sure I understood.
Any bucket / prefix is valid (as long as they can write to it, but that's another issue that we can't solve here).

`s3://my-bucket/prefix` is valid

Reply:
Well then why not construct the URL infallibly?
```suggestion
	s3URI := &url.URL{Scheme: "s3", Host: s3Bucket, Path: "/" + s3Prefix}
```

Reply:
There's still https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html#general-purpose-bucket-names that we could use for bucket name validity.

Reply:
Yeah, you are right.
I was trying to include some validation before accepting the URL.
I though that just setting the `url.Path` would possibly create an invalid URL, but it doesn't seem to be the case.
I'll change it as you suggested, thank you.